Library/Homebrew/language/haskell.rb

```diff
@@ -92,5 +92,50 @@ def install_cabal_package(*args, **options)
         end
       end
     end
+
+    module CabalV2
+      def cabal_sandbox(options = {})
+        pwd = Pathname.pwd
+        home = options[:home] || pwd
+
+        # pretend HOME is elsewhere, so that ~/.cabal is kept as untouched
+        # as possible (except for ~/.cabal/setup-exe-cache)
+        # https://github.com/haskell/cabal/issues/1234
+        saved_home = ENV["HOME"]
+        ENV["HOME"] = home
+
+        system "cabal", "v2-update"
+
+        yield
+
+        # restore the environment
+        ENV["HOME"] = saved_home
+      end
+
+      def cabal_install(*args)
+        # cabal hardcodes 64 as the maximum number of parallel jobs
+        # https://github.com/Homebrew/legacy-homebrew/issues/49509
+        make_jobs = (ENV.make_jobs > 64) ? 64 : ENV.make_jobs
+
+        # cabal-install's dependency-resolution backtracking strategy can easily
+        # need more than the default 2,000 maximum number of "backjumps," since
+        # Hackage is a fast-moving, rolling-release target. The highest known
+        # needed value by a formula at this time (February 2016) was 43,478 for
+        # git-annex, so 100,000 should be enough to avoid most gratuitous
+        # backjumps build failures.
+        system "cabal", "v2-install", "--jobs=#{make_jobs}", "--max-backjumps=100000", *args
```

Comment:
It might be nicer to have a method on `Formula` that provides the sensible arguments here instead of `cabal_install`. See `std_cmake_args`; could be `std_cabal_v2_install_args`

Reply:
I agree, it would be nicer.  The problem is that `cmake` is designed such that you *can* run a single command that Does The Right Thing (assuming the right command line parameters).  `cabal` need a little babysitting first, for example running `cabal update`, and then either by redirecting `$HOME`, or [by setting `$CABAL_DIR`](https://github.com/haskell/cabal/issues/6517).

Reply:
I think having it be two separate commands would be nice/preferable to a DSL that hides the commands (as the existing one does) and is much harder to debug as a result.